src/Symfony/Component/JsonPath/JsonCrawler.php

```diff
@@ -29,25 +36,29 @@
  */
 final class JsonCrawler implements JsonCrawlerInterface
 {
-    private const RFC9535_FUNCTIONS = [
-        'length' => true,
-        'count' => true,
-        'match' => true,
-        'search' => true,
-        'value' => true,
-    ];
-
     private const COMPARISON_OPERATORS = ['!=', '==', '>=', '<=', '>', '<'];
 
+    private readonly JsonPathFunctionsProviderInterface $functionsProvider;
+
     /**
      * @param resource|string $raw
      */
     public function __construct(
         private readonly mixed $raw,
+        ?JsonPathFunctionsProviderInterface $functionsProvider = null,
     ) {
         if (!\is_string($raw) && !\is_resource($raw)) {
             throw new InvalidArgumentException(\sprintf('Expected string or resource, got "%s".', get_debug_type($raw)));
         }
+
+        $this->functionsProvider = new JsonPathFunctionsProvider([
+            new LengthFunction(),
+            new CountFunction(),
+            new MatchFunction(),
+            new SearchFunction(),
+            new ValueFunction(),
+            ...$functionsProvider?->getFunctions() ?? [],
```

Comment:
This also breaks any lazy-loading of the function.

To combine a ServiceLocator-based injection of custom functions with core functions, you might want to create a combined provider, or something like that.